src/include/uapi/ipc/control.h

```diff
@@ -152,4 +152,21 @@ struct sof_ipc_ctrl_data {
 	};
 } __attribute__((packed));
 
+/* event type */
+enum sof_ipc_ctrl_event_type {
+	SOF_CTRL_EVENT_KD_NOTIFICATION = 0,	/**< keyword detection event */
+};
+
+struct sof_ipc_comp_event {
+	struct sof_ipc_reply rhdr;
+	uint32_t src_comp_id;	/**< source component id */
+	uint32_t type;	/**< event type - SOF_CTRL_EVENT_* */
+	/**< control data - add new types if needed */
+	union {
+		/**< data can be used by binary controls */
+		struct sof_abi_hdr data[0];
+		/**< enum values i.e. SUCCESS, ERROR etc. */
+		uint32_t event_value;
```

Comment:
we should link event_value to an explicit enum, otherwise leave it completely opaque. this is in-between and unclear if you can add your own codes or need to follow a list shared with the kernel.

Reply:
also remind me if there is any field describing the size of the data?

Reply:
structure updated